### ISSUE TITLE
Installation test with setup.py fails on Python 3.7 with bad marshaling data

### ISSUE THREAD

**Issue by andy-maier** [pywbem/pywbem#2167]
The install test using `python setup.py install` in the repo work directory fails when installing the first dependent package.

Environment:
* Travis with Python 3.7 with minimum package levels
* stable_0.17 branch

See this Travis run on Python 3.7 with minimum package levels: https://travis-ci.org/github/pywbem/pywbem/jobs/674160741:

```
Processing dependencies for pywbem==0.17.1
Searching for six>=1.10.0; python_version < "3.8"
Reading https://pypi.python.org/simple/six/
Downloading https://files.pythonhosted.org/packages/21/9f/b251f7f8a76dec1d6651be194dfba8fb8d7781d10ab3987190de8391d08e/six-1.14.0.tar.gz#sha256=236bdbdce46e6e6a3d61a337c0f8b763ca1e8717c03b369e87a7ec7ce1319c0a
Best match: six 1.14.0
Processing six-1.14.0.tar.gz
Writing /tmp/easy_install-ejwc9evr/six-1.14.0/setup.cfg
Running six-1.14.0/setup.py -q bdist_egg --dist-dir /tmp/easy_install-ejwc9evr/six-1.14.0/egg-dist-tmp-l8sq1x8l
/opt/python/3.7.1/lib/python3.7/distutils/dist.py:274: UserWarning: Unknown distribution option: 'long_description_content_type'
  warnings.warn(msg)
/opt/python/3.7.1/lib/python3.7/distutils/dist.py:274: UserWarning: Unknown distribution option: 'project_urls'
  warnings.warn(msg)
no previously-included directories found matching 'documentation/_build'
zip_safe flag not set; analyzing archive contents...

Traceback (most recent call last):
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/sandbox.py", line 157, in save_modules
    yield saved
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/sandbox.py", line 198, in setup_context
    yield
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/sandbox.py", line 255, in run_setup
    DirectorySandbox(setup_dir).run(runner)
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/sandbox.py", line 285, in run
    return func()
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/sandbox.py", line 253, in runner
    _execfile(setup_script, ns)
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/sandbox.py", line 47, in _execfile
    exec(code, globals, locals)
  File "/tmp/easy_install-ejwc9evr/six-1.14.0/setup.py", line 57, in <module>
    if not line.startswith('#') and line != '':
  File "/opt/python/3.7.1/lib/python3.7/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/opt/python/3.7.1/lib/python3.7/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/opt/python/3.7.1/lib/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/command/bdist_egg.py", line 209, in run
    os.path.join(archive_root, 'EGG-INFO'), self.zip_safe()
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/command/bdist_egg.py", line 245, in zip_safe
    return analyze_egg(self.bdist_dir, self.stubs)
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/command/bdist_egg.py", line 355, in analyze_egg
    safe = scan_module(egg_dir, base, name, stubs) and safe
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/command/bdist_egg.py", line 392, in scan_module
    code = marshal.load(f)
ValueError: bad marshal data (unknown type code)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "setup.py", line 138, in <module>
    'Topic :: System :: Systems Administration',
  File "/opt/python/3.7.1/lib/python3.7/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/opt/python/3.7.1/lib/python3.7/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/opt/python/3.7.1/lib/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/command/install.py", line 67, in run
    self.do_egg_install()
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/command/install.py", line 117, in do_egg_install
    cmd.run()
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/command/easy_install.py", line 411, in run
    self.easy_install(spec, not self.no_deps)
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/command/easy_install.py", line 655, in easy_install
    return self.install_item(None, spec, tmpdir, deps, True)
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/command/easy_install.py", line 702, in install_item
    self.process_distribution(spec, dist, deps)
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/command/easy_install.py", line 747, in process_distribution
    [requirement], self.local_index, self.easy_install
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/pkg_resources/__init__.py", line 846, in resolve
    dist = best[req.key] = env.best_match(req, ws, installer)
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1118, in best_match
    return self.obtain(req, installer)
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1130, in obtain
    return installer(requirement)
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/command/easy_install.py", line 674, in easy_install
    return self.install_item(spec, dist.location, tmpdir, deps)
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/command/easy_install.py", line 700, in install_item
    dists = self.install_eggs(spec, download, tmpdir)
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/command/easy_install.py", line 881, in install_eggs
    return self.build_and_install(setup_script, setup_base)
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/command/easy_install.py", line 1120, in build_and_install
    self.run_setup(setup_script, setup_base, args)
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/command/easy_install.py", line 1106, in run_setup
    run_setup(setup_script, args)
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/sandbox.py", line 258, in run_setup
    raise
  File "/opt/python/3.7.1/lib/python3.7/contextlib.py", line 130, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/sandbox.py", line 198, in setup_context
    yield
  File "/opt/python/3.7.1/lib/python3.7/contextlib.py", line 130, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/sandbox.py", line 169, in save_modules
    saved_exc.resume()
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/sandbox.py", line 144, in resume
    six.reraise(type, exc, self._tb)
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/pkg_resources/_vendor/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/sandbox.py", line 157, in save_modules
    yield saved
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/sandbox.py", line 198, in setup_context
    yield
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/sandbox.py", line 255, in run_setup
    DirectorySandbox(setup_dir).run(runner)
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/sandbox.py", line 285, in run
    return func()
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/sandbox.py", line 253, in runner
    _execfile(setup_script, ns)
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/sandbox.py", line 47, in _execfile
    exec(code, globals, locals)
  File "/tmp/easy_install-ejwc9evr/six-1.14.0/setup.py", line 57, in <module>
    if not line.startswith('#') and line != '':
  File "/opt/python/3.7.1/lib/python3.7/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/opt/python/3.7.1/lib/python3.7/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/opt/python/3.7.1/lib/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/command/bdist_egg.py", line 209, in run
    os.path.join(archive_root, 'EGG-INFO'), self.zip_safe()
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/command/bdist_egg.py", line 245, in zip_safe
    return analyze_egg(self.bdist_dir, self.stubs)
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/command/bdist_egg.py", line 355, in analyze_egg
    safe = scan_module(egg_dir, base, name, stubs) and safe
  File "/home/travis/build/pywbem/pywbem/tmp_installtest/virtualenvs/pywbem_test_test2/lib/python3.7/site-packages/setuptools/command/bdist_egg.py", line 392, in scan_module
    code = marshal.load(f)
ValueError: bad marshal data (unknown type code)
```

**Comment by andy-maier** [comment]
This was a bug in setuptools that was fixed in setuptools 38.4.1. See https://github.com/pypa/setuptools/issues/1257.

Solution is to increase the minimum version of setuptools on Python 3.7, which is currently 33.1.1 (see minimum-constraints.txt).

**Comment by andy-maier** [comment]
Fixed in PR #2169 for master and #2168 for 0.17.1.
